src/eddington/fit_data.py

```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
             return True
         except ValueError:
             return False
+
+    def __make_sheet(self, sheet=None):
```

Comment:
There is no separate logic here that may be called in more than one place.

You can move the entire section to the `save_excel` method